### PR TITLE
Add some time-getting functions

### PR DESCRIPTION
**Pull request by IvanGrigorik**
# Time module
Change the name of [`time_now_day`](https://github.com/EngineeringSoftware/gobash/blob/c86905e1a66b4a09eab8ed695f074aed73b76752/src/util/time.sh#L30) function to [`time_now_day_of_week`](https://github.com/IvanGrigorik/gobash/blob/4a8692a0599ceeae0af1d05f1d1b4a11e83e7b89/src/util/time.sh#L30) with corresponding comment and [`test`](https://github.com/IvanGrigorik/gobash/blob/e9945ba9dcd57307806a32b23b07b3c56193ae02/src/util/time_test.sh#L25).

Add some time-getting functions: 
- [`time_now_day_of_week_str`](https://github.com/IvanGrigorik/gobash/blob/4a8692a0599ceeae0af1d05f1d1b4a11e83e7b89/src/util/time.sh#L39) : get day of the week name as string
- [`time_now_day_of_month`](https://github.com/IvanGrigorik/gobash/blob/4a8692a0599ceeae0af1d05f1d1b4a11e83e7b89/src/util/time.sh#L48) : get current day of the month
- [`time_now_month_str`](https://github.com/IvanGrigorik/gobash/blob/4a8692a0599ceeae0af1d05f1d1b4a11e83e7b89/src/util/time.sh#L75) : get month name as string

with corresponding tests:
- [`test_time_now_day_of_week_str`](https://github.com/IvanGrigorik/gobash/blob/e9945ba9dcd57307806a32b23b07b3c56193ae02/src/util/time_test.sh#L34)
- [`test_time_now_day_of_month`](https://github.com/IvanGrigorik/gobash/blob/e9945ba9dcd57307806a32b23b07b3c56193ae02/src/util/time_test.sh#L52)
- [`test_time_now_month_str`](https://github.com/IvanGrigorik/gobash/blob/e9945ba9dcd57307806a32b23b07b3c56193ae02/src/util/time_test.sh#L77)

### CODE REVIEW

File: src/util/time_test.sh [hunk 1]
```diff
@@ -22,14 +22,41 @@ function test_now_millis() {
...
+        *) assert_fail "non-existing day of the week";;
+        esac
+}
+readonly -f test_time_now_day_of_week_str
+
+function test_time_now_day_of_month() {
+        local day
+        day=$(time_now_day_of_month) || \
+                assert_fail
+        assert_gt "${day}" 0
```
Comment (gliga) [inline src/util/time_test.sh]: the only change I would make is on this line and the follow line
```
[ "${day}" -lt 0 ] && assert_fail
[ "${day}" -gt 32 ] && assert_fail
return 0
```

**Review by gliga — APPROVED** [review]
One comment to address; the rest looks good.

File: src/util/time_test.sh (hunk 1 above)
Comment (IvanGrigorik) [reply src/util/time_test.sh]: I can add it, but why if we have `assert_gt` and `assert_lt` methods, and the whole file uses these methods, for example in the [`test_time_now_day_of_week`](https://github.com/IvanGrigorik/gobash/blob/e9945ba9dcd57307806a32b23b07b3c56193ae02/src/util/time_test.sh#L25-L30) function?

Maybe then I should change all occurrences in this file?

Comment (gliga) [reply src/util/time_test.sh]: Great point and question.
`assert_` functions are legacy code before we decided to make our library fully compatible with `set -e` (i.e., if invokes test directly as a function the execution should not stop). Documented here: https://engineeringsoftware.github.io/gobash/testing.html#asserting-results

You are correct that all those assert should be changed in this file. If you have time, go ahead. If not, you can only update your own change. Thanks for the contribution in either case.

Comment (IvanGrigorik) [reply src/util/time_test.sh]: Updated.

**Review by gliga — APPROVED** [review]
LG

**Pull request merged**